src/main/java/core/basesyntax/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.Inventory;
+
+public class SupplyOperation implements FruitOperationHandler {
+    @Override
+    public void executeOperation(FruitTransaction transaction, Inventory inventory) {
+        inventory.getInventory().put(transaction.getFruit(),
```

Comment:
Avoid directly accessing and modifying the inventory's collection. Instead, consider providing a method in the `Inventory` class to abstract these details away. This will help in maintaining encapsulation.